src/app/event-system/services/fixed-interval-reconnect-strategy.service.ts

```diff
@@ -0,0 +1,51 @@
+import { ReconnectStrategy } from '../abstractions/reconnect-strategy.service'
+import { Logger } from '../../core/abstractions/logger.service'
+
+const RECONNECT_INTERVAL_IN_MS: number = 2000
+
+export class FixedIntervalReconnectStrategy implements ReconnectStrategy {
+  private readonly logger: Logger
+  private connectAttempts: number = 0
+  private delayTimer?: NodeJS.Timeout
+
+  constructor(logger: Logger) {
+    this.logger = logger.tag('FixedIntervalReconnectStrategy')
+  }
+
+  public connected(): void {
+    this.connectAttempts = 0
+  }
```

Comment:
Also, I don't see you resetting `connectAttemps`? Is that on purpose?

Reply:
I have removed connectAttempts from the class.
connected is void, so it will never print true.
`.connected` and `.disconnected` are methods for informing the reconnect strategy of these events, such that it can react to them.
Suggestions for better names are appreciated.